bot/extensions/vote_cog.py

```diff
@@ -0,0 +1,315 @@
+			await ctx.channel.send(f'{win_emoji} ***option has won!***')
+		else:
+			await ctx.channel.send('No one voted.')
+
+
+
+	def build(self) -> None:
+		# Set the embed title
+		self.title = self._poll_title
+		
+		# Generate the description
+		main_text = ''
+		if self._timer_label:
+			main_text += (self._timer_label + '\n\n')
+
+		for emoji_index, option in enumerate(self._poll_options):
+			# Check if the number of emojis needed is not out of boundaries
+			if emoji_index < len(self._poll_emojis):
+				emoji = self._poll_emojis[emoji_index]
+			else:
+				break
+			main_text += (emoji + ' **' + option + '**: ' + str(self._poll_counter[emoji]) + '\n\n')
+
+		self.description = main_text
+
+class VoteButton(Button):
+	def __init__(self, emoji) -> None:
+		super().__init__(style=discord.ButtonStyle.gray, emoji=emoji)
+
+	def set_embed_reference(self, embed: PollEmbed) -> None:
+		self._embed = embed
+
+
+	async def callback(self, interaction: discord.Interaction) -> None:
+		if self._embed is None:
+			raise ValueError('Update embed function or/and Counter reference are not set.')
+		# TODO: Catch the unknown interaction error and send a message that user is clicking too fast.
+		if self._embed.user_voted(interaction.user):
+			user_emoji = self._embed.get_user_emoji(interaction.user)
+			if self.emoji.name != user_emoji:
+				self._embed.decrement_counter(user_emoji)
+				self._embed.increment_counter(self.emoji.name)
+				self._embed.set_user(interaction.user, self.emoji.name)
+				self._embed.build()
+				await interaction.message.edit(embed=self._embed)
+		else:
+			self._embed.increment_counter(self.emoji.name)
+			self._embed.set_user(interaction.user, self.emoji.name)
+			self._embed.build()
+			await interaction.message.edit(embed=self._embed)
+		await interaction.response.defer()
+
+'''
+	PollEmbed:
+		counter = {}
+	ButtonsView:
+		VoteButton -> (emoji)
+		VoteButton.set_embed_ref(PollEmbed)
+		VoteButton.callback() -> counter[emoji] += 1; embed_update();
+'''
+class Timer:
+	def __init__(self, msg: discord.Message, seconds: int, embed: PollEmbed, view: PollView) -> None:
+		self._seconds = seconds
+		self._embed = embed
+		self._view = view
+		self._msg = msg
+
+	async def start(self) -> None:
+		while self._seconds >= 0:
+			await self.timer_info_update()
+			await asyncio.sleep(1)
+			self._seconds -= 1
+
+		self._view.disable_buttons()
+		await self._msg.edit(embed=self._embed, view=self._view)
+
+
+	async def timer_info_update(self) -> None:
+		if self._seconds // 60 >= 10:
+			if self._seconds % 60 >= 10:
+				self._embed.set_timer_label(f'**{self._seconds // 60}:{self._seconds % 60}**')
+			else:
+				self._embed.set_timer_label(f'**{self._seconds // 60}:0{self._seconds % 60}**')
+		else:
+			if self._seconds % 60 >= 10:
+				self._embed.set_timer_label(f'**0{self._seconds // 60}:{self._seconds % 60}**')
+			else:
+				self._embed.set_timer_label(f'**0{self._seconds // 60}:0{self._seconds % 60}**')
```

Comment:
I see those two lines repeated twice in your code, could be worth creating a @staticmethod instead of being redundant and using the function every time you need it?